### PR TITLE
[2.x] Make dataset method return values by name

### PR DESCRIPTION
**Pull request by hafezdivandari**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| Fixed tickets | -

This PR makes it possible to get the specified dataset values by its name:

```php
dataset('foo', [1, 2]);

test('can get dataset values', function () {
    expect(dataset('foo'))->toBe([1, 2]);
});
```

### CODE REVIEW

File: src/Functions.php [hunk 1]
```diff
@@ -57,10 +57,15 @@ function beforeEach(Closure $closure = null): BeforeEachCall
     /**
      * Registers the given dataset.
      *
-     * @param  Closure|iterable<int|string, mixed>  $dataset
+     * @param  Closure|iterable<int|string, mixed>|null  $dataset
+     * @return array|void
      */
-    function dataset(string $name, Closure|iterable $dataset): void
+    function dataset(string $name, Closure|iterable $dataset = null)
```
Comment (paulshryock) [inline src/Functions.php:63]: Should this still have a return type hint?

```php
function dataset(string $name, Closure|iterable $dataset = null): array|void
```

Comment (hafezdivandari) [reply src/Functions.php:63]: Void can only be used as a standalone type.

Comment (paulshryock) [reply src/Functions.php:63]: Of course; my bad.

**Comment by nunomaduro** [comment]
No plans for this.

**Pull request closed without merging**